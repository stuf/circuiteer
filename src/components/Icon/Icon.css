.icon {
  @apply w-6 h-6;
}

.icon,
.icon__el {
  @apply w-6 h-6;
}

.icon--ui {
  /* @apply border-green-500; */
}

.icon__el {
  /* @apply border border-blue-500; */
}
