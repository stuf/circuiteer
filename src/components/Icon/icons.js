export { Reorder } from '@material-ui/icons';
