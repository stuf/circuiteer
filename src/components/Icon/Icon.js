import * as P from 'prop-types';
import * as icons from './icons';

import './Icon.css';

export function Icon(props) {
  const { name } = props;

  const _icon = icons[name];

  if (!_icon) {
    console.log('Invalid icon %s', name, { icons });
    return <div>Invalid</div>;
  }

  const IconSvg = _icon;

  return (
    <span className="icon icon--ui">
      <IconSvg className="icon__el" />
    </span>
  );
}

Icon.propTypes = {
  wrap: P.bool,
  nowrap: P.bool,
  name: P.string,
  size: P.number,
};

Icon.defaultProps = {
  wrap: true,
  size: 24,
};
