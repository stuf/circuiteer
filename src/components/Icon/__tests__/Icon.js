import { render } from '@testing-library/react';
import { Icon } from '../Icon';

describe('Icon', () => {
  test('base case', async () => {
    const props = {
      name: 'settings',
    };

    const { container } = render(<Icon {...props} />);

    expect(container).toMatchSnapshot();
  });

  test('custom size', async () => {
    const props = {
      name: 'settings',
      size: 16,
    };

    const { container } = render(<Icon {...props} />);

    expect(container).toMatchSnapshot();
  });
});
