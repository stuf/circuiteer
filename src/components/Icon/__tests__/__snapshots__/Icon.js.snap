// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon base case 1`] = `
<div>
  <span
    class="icon icon--uiflex items-center justify-center"
    style="width: 24px; height: 24px;"
  >
    <span
      class="material-icons material-icons-outlined"
      style="font-size: 24px;"
    >
      settings
    </span>
  </span>
</div>
`;

exports[`Icon custom size 1`] = `
<div>
  <span
    class="icon icon--uiflex items-center justify-center"
    style="width: 16px; height: 16px;"
  >
    <span
      class="material-icons material-icons-outlined"
      style="font-size: 16px;"
    >
      settings
    </span>
  </span>
</div>
`;
